web/src/context/l10n.jsx

```diff
@@ -146,7 +146,27 @@ function navigatorLanguages() {
  */
 function findSupportedLanguage(languages) {
   const supported = Object.keys(cockpit.manifests.agama?.locales || {});
-  return languages.find(lang => supported.includes(lang));
+
+  for (const candidate of languages) {
+    const match = supported.find(s => s.startsWith(candidate));
```

Comment:
In theory there might be 3 letter languages, after selecting "Cherokee" in Chrome it returns `'chr'` in the `navigator.languages` list. I'd rather not rely on the fact that all language codes have the same length.

I'd rather split the locale with `-` and compare the parts. It is more complex, but more reliable.